src/main/java/me/namuhuchutong/ladder/controller/LadderController.java

```diff
@@ -1,47 +1,48 @@
 
-import me.namuhuchutong.ladder.domain.*;
-import me.namuhuchutong.ladder.domain.factory.ScaffoldFactory;
-import me.namuhuchutong.ladder.dto.LadderResult;
-import me.namuhuchutong.ladder.dto.NameAndResult;
-import me.namuhuchutong.ladder.dto.UserInputInformation;
+import me.namuhuchutong.ladder.domain.engine.Ladder;
+import me.namuhuchutong.ladder.domain.engine.LadderCreator;
+import me.namuhuchutong.ladder.domain.engine.dto.LadderResult;
+import me.namuhuchutong.ladder.domain.engine.dto.UserInputInformation;
 import me.namuhuchutong.ladder.ui.InputView;
 import me.namuhuchutong.ladder.ui.OutputView;
 
+import static me.namuhuchutong.ladder.beans.FactoryBean.*;
+
 public class LadderController {
 
-    private final InputView inputView;
+    private static final String ALL = "all";
 
-    private final ScaffoldFactory scaffoldFactory;
+    private final InputView inputView;
 
     private final OutputView outputView;
 
-    public LadderController(InputView inputView, ScaffoldFactory scaffoldFactory, OutputView outputView) {
+    public LadderController(InputView inputView, OutputView outputView) {
         this.inputView = inputView;
-        this.scaffoldFactory = scaffoldFactory;
         this.outputView = outputView;
     }
 
     public void start() {
         UserInputInformation information = new UserInputInformation(inputView.inputStringNames(),
                                                                     inputView.inputStringResults(),
                                                                     inputView.inputInteger());
-        Names names = Names.from(information.getInputNames());
-        Rows rows = Rows.createLadder(information.getLadderHeight(),
-                                      names.size(),
-                                      scaffoldFactory);
-        Results results = Results.from(information.getInputResults());
-        outputView.printResult(new LadderResult(names, rows, results));
-        NameAndResult nameAndResult = new Ladder(rows, names, results).startLadderGame();
-        showEachResult(names, nameAndResult);
-        outputView.printNameAndResult(nameAndResult.getAllResults());
+        LadderCreator ladderCreator = ladderCreator();
+        Ladder ladder = ladderCreator.createLadder(information);
+        LadderResult ladderResult = ladder.startLadderGame();
+        outputView.printResult(ladderResult.showLadder());
+        showEachResult(ladderResult);
+        outputView.printNameAndResult(ladderResult.getAllResults());
     }
 
-    private void showEachResult(Names names, NameAndResult nameAndResult) {
-        String inputName = inputView.inputStringName();
-        while (names.contains(inputName)) {
-            outputView.printNameAndResult(nameAndResult.getResult(inputName));
-            inputName = inputView.inputStringName();
+    private void showEachResult (LadderResult ladderResult) {
+        String input = inputView.inputStringName();
+        while (isNotAll(input)) {
```

Comment:
```suggestion
        Name name = ..;
        while (name.isAll()) {
```

`"all"을 입력하면 전체 참여자의 실행 결과를 출력한다.` 기능 요구사항을 중요한 도메인 로직이라고 바라보는 관점으로 보았을때, 도메인 객체에게 물어볼수 있을것 같아요 😄 
(물론 지금 설계하신 흐름과 안맞을수 있을것 같긴하네요 😂 )